### DISCUSSION TITLE
Is there a way to delete uploaded images using CldUploadWidget

### DISCUSSION THREAD

**Discussion by vikmehta** [cloudinary-community/next-cloudinary#221] [Q&A] [answered]
In the **CldUploadWidget** component, we get the access to Cloudinary instance

```
<CldUploadWidget>
  {({ cloudinary, widget, open }) => {
    // UI
  }}
</CldUploadWidget>
```
Is it somehow possible to delete an uploaded asset using this cloudinary instance?
Something like this:

```
<CldUploadWidget>
    {({ cloudinary, widget, open }) => {
        const handleDeleteImage = async (id) => {
            await cloudinary.uploader.destroy(id)
        }

        return (
            <DeleteButton onClick={() => handleDeleteImage(id)} />
        )
    }}
</CldUploadWidget>
```

Or Is there a way to delete uploaded images using CldUploadWidget or any other components from next-cloudinary?
I couldn't find any documentation on deleting an asset using next-cloudinary package.
Please share your thoughts on how can one delete an uploaded image; If not using this package then any other alternative that allows Cloudinary image deletion from NextJs

**Comment by colbyfayock** [comment, marked as answer]
@vikmehta unfortunately there's not currently a native way in Next Cloudinary to delete an asset

you would likely need to set up an API Endpoint (or Server Action in Next.js 13 App Router) which makes a request to the Admin API to delete the asset you want

https://cloudinary.com/documentation/admin_api#delete_resources

**Reply by vikmehta** [reply]
Thank you so much for your reply @colbyfayock  I have watched most of your videos on Cloudinary. They are really helpful. I also love this next-cloudinary package that makes it super simple to upload and display images from Cloudinary.

**Comment by ShueiYang** [comment]
Hello,
If it's not support, does this mean I will still need to install the standard SDK cloudinary in order to make an API Endpoint to delete Image in Nextjs ? then what's the point to use this package if I can upload / update anyway in the native cloudinary ?

**Reply by colbyfayock** [reply]
This library is largely client driven. You would need to use the Node.js SDK serverside or hit the REST API manually via the Admin API in order to delete an asset along with your API Key and Secret.

**Reply by ShueiYang** [reply]
Ok thanks for the answer 👍
